composer/trainer/trainer.py

```diff
@@ -180,7 +180,91 @@ def _get_initial_grad_accum(grad_accum: Union[int, str]):
 
 def _is_cuda_oom(e: RuntimeError):
     """Determines if error is CUDA Out of Memory and if adaptive_grad_accum is enabled."""
-    return 'CUDA out of memory' in str(e)
+    if 'CUDA out of memory' in str(e):
+        return True
+    # With batch_norm, large batch sizes sometimes result in cuDNN instead of Cuda OOMs.
+    if 'cuDNN error: CUDNN_STATUS_NOT_SUPPORTED. This error may appear if you passed in a non-contiguous input.' in str(
+            e):
+        warnings.warn(
+            f'Encountered "cuDNN error: CUDNN_STATUS_NOT_SUPPORTED. This error may appear if you passed in a non-contiguous input." This can happen when the batch_size is too large for the GPU so auto grad_accum will rerun with a higher grad_accum value, but there may be a user error with non-contiguous inputs.'
+        )
+        return True
```

Comment:
Is there a better way to handle this?

Reply:
This error isn't specific to CUDA OOM, so I am concerned about catching non-OOM errors with this check. Did we encounter use cases with this, and how common is this?

Reply:
I'm encountering this consistently with resnet50 on a100 with in batchnorm with 8k batch size. I'm happy to skip though since its an edge case

Reply:
OK I'm fine keeping it in, that seems like a reasonable case.